containerd-shim/main.go

```diff
@@ -100,6 +101,9 @@ func start(log *os.File) error {
 			msgC <- m
 		}
 	}()
+	if runtime.GOOS == "solaris" {
+		return nil
```

Comment:
On Solaris we do not close the pipes in p.create() and block on the call to runz create itself. Therefore we'll get to this piece of code only after the container exits. This was our workaround to not being able to use the subreaper functionality.